File: rot13/rot13.erl
-module(rot13).
-export([rot13/1]).

rot([]) ->
    io:format(" ");
rot([Letter|Tail]) ->
    if
        $a =< Letter, Letter =< $m -> io:format("~c", [Letter+13]);
        $n =< Letter, Letter =< $z -> io:format("~c", [Letter-13]);
        $A =< Letter, Letter =< $M -> io:format("~c", [Letter+13]);
        $N =< Letter, Letter =< $Z -> io:format("~c", [Letter-13]);
        true -> io:format("~c", [Letter])
    end,
    rot(Tail).

rot13([]) -> 
    io:format("~n"),
    init:stop();
rot13([Arg|Tail]) ->
    rot(Arg),
    rot13(Tail).

% TAIL INFO:
% Name: ROT13
% Language: Erlang
% Compile: erlc rot13.erl
% State: Done
%
% Rotates a string
%
%
% Example: erl -noshell -run rot13 rot13 Hello World \!
%

File: rot13/rot13.hs
module Main where

import System.Environment
import Data.Char

rot13 :: Char -> Char
rot13 c
  | 'a' <= toLower c && toLower c <= 'm' = chr(ord c + 13)
  | 'n' <= toLower c && toLower c <= 'z' = chr(ord c - 13)
  | otherwise = c

main = do
  arg <- System.Environment.getArgs
  if length arg > 0 then
    putStrLn $ map rot13 $ unwords arg
    else
    interact $ map rot13
    
-- TAIL INFO:
-- Name: Rot13
-- Language: Haskell
-- Compile: ghc --make rot13.hs
-- State: Done
--
-- Rotate the argument by 13 letters
--
-- Example: ./rot13 hello world
-- Example2: cat rot13.hs | ./rot13
--
